### ISSUE TITLE
Dev tools states shows friendly_name instead of entity_id

### ISSUE THREAD

**Issue by Mariusthvdb** [home-assistant/frontend#11938]
### Checklist

- [X] I have updated to the latest available Home Assistant version.
- [X] I have cleared the cache of my browser.
- [X] I have tried a different browser to see if it is related to my browser.

### Describe the issue you are experiencing

<img width="616" alt="Schermafbeelding 2022-03-04 om 23 24 22" src="https://user-images.githubusercontent.com/33354141/156850068-3c4d6a22-f363-4e22-8565-4c0525f8d31b.png">

as title. We go to the dev tools pages to be able to check and confirm our state and entity_id's and showing the friendly_name in the entity field does not help. Please return to showing entity_id. Friendly_name is also shown in the details list, so very easy to check on that anyways. 
We have that twice, and no entity_id.

Please note we can have multiple entities with the same friendly_name: sensor, binary_sensor, input_boolean, heck, even automation.... we need to be able to see and differentiate in 1 glance as much as possible.

thanks for considering

### Describe the behavior you expected

show  entity_id like before

### Steps to reproduce the issue

1. nothing special, just navigate to the dev tools
2.
3.
...

### What version of Home Assistant Core has the issue?

2022.3.X

### What was the last working version of Home Assistant Core?

2022.2

### In which browser are you experiencing the issue with?

Any

### Which operating system are you using to run this browser?

MacOS  Monterey 12.2.1

**Comment by ildar170975** [comment]
Another example:
![изображение](https://user-images.githubusercontent.com/71872483/156859212-76396c27-e140-4a8d-a582-c2babf638261.png)
Can you guess what “entity_id” is displayed here?
Vote for displaying "entity_id" instead of "friendly_name".

**Comment by nickrout** [comment]
What appears there does seem to depend on how the entity is chosen (which IMHO is undesirably inconsistent)

Option 1: search for the entity at the top of the page in the set state dialog, and the friendly name is shown

Option 2: search for the entity in "Filter Entities" in the lower panel, then click on the entity name, the old behaviour is shown and the entity_id appears in the "Set State" dialog above UNLESS the entity has first been chosen in Option 1.

**Comment by parautenbach** [comment]
This seems like something for the UX person that joined the team a while ago to have a good look at. I'm a principled person, so I'd say (as a design principle): Information in a dev (debug) tool should be complete and at a low level to be useful. The friendly name is a property of the entity and in this case shouldn't be the primary method to identify something. Having an option to look up by friendly name is useful, but at its core, the entity ID is what connects everything. Friendly names aren't unique. Entity IDs are.

**Comment by ildar170975** [comment]
I agree that Dev tools should be low level.
Sometimes I have to post a screenshot of the “set state” area in the Community to get a technical support. The more information is provided the better.

**Comment by emufan** [comment]
Can only second this. First the hiding/dropdown in the last release, now this one. The dev-tools should really stay useful and not with all this, what is recently done.

**Comment by ildar170975** [comment]
Same useless changes for Automations: https://github.com/home-assistant/frontend/issues/11870

![изображение](https://user-images.githubusercontent.com/71872483/156902375-ef53b1d7-6ca2-4123-a7ca-3a9f3f331ecb.png)

**Comment by RoboMagus** [comment]
It's not just dev-tools. Similar with History, and it's really ~~F-ing~~ annoying! 
There should be some place that holds the entity_id so we can find the reference for the thing that needs to be manipulated e.g. in yaml.

I would happily call this a bug, but I fear that the devs have made the explicit choice to implement this as a feature. (hope I'm wrong though).

Regardless of what's supposed to end up in those boxes, the fact that it auto-completes `entity_id`'s and then just replaces it with the friendly name is not a great user experience.

**Comment by Mariusthvdb** [comment]
one might eve say this extends to the Areas view also:

<img width="399" alt="Schermafbeelding 2022-03-07 om 15 32 37" src="https://user-images.githubusercontent.com/33354141/157054123-7cdb830a-3b90-4721-b105-51f944a5a496.png">

where one would believe these to be media_players, but they are a switch and a sensors.. or

<img width="349" alt="Schermafbeelding 2022-03-07 om 15 32 50" src="https://user-images.githubusercontent.com/33354141/157054222-c77d094d-6408-4ee1-8529-ffa8148b8dd4.png">

for device, media_player and google_cast.

**Comment by RoboMagus** [comment]
@zsarnett, Since there are minor fixes being pushed through in some intermediate releases. 
Any chance this will be picked up soon too? 
This is clearly an impractical change.

**Comment by OzGav** [comment]
I’m adding my voice to those concerned about this change. @nickrout makes a very good point about the inconsistency of the behaviour on the dev tools page. Whilst I now know how to get the info on this page that I want (by searching in the filter entities box) there are many other examples posted by others above that show there is an issue here.
Link to my forum post https://community.home-assistant.io/t/developer-tools-no-longer-shows-entity-id/408581/2

**Comment by PolarLava** [comment]
I have to agree that this has made life unnecessarily more difficult for those of us who live in the YAML config world.  Previously you could cut-and-paste entity_id's and you've taken that option away.  If we can't go back to the way it was can we at least compromise and have both "Friendly Name" and "Entity ID" fields?  At least that way everyone gets what they want or need here.

The additional step of having to click "Set State" in order to enter and entity is also a nuisance, but not nearly as bad as this change.  Again, if we can't have the previous setup where the Entity field was displayed on page load, can we at least have a compromise?  Either a configuration variable that sets "Set State" open or closed by default or, once I click "Set State" it stays open?  This may make the interface cleaner, but as a debugging took it's just adding unnecessary overhead.

**Comment by OzGav** [comment]
Good point on the additional step. I also find that tedious.

**Comment by Mariusthvdb** [comment]
while we can search on the actual entity_id in dev tools state, I now discovered we can NOT do that in config/script/dashboard and related pages.

Debugging my yaml script, and copying several script id's on this search bar rendered no result.

<img width="1159" alt="Schermafbeelding 2022-04-10 om 14 31 44" src="https://user-images.githubusercontent.com/33354141/162618280-63cd173e-73bc-4516-9486-fbc15a0a4c1e.png">

and we Must use the Name:
<img width="1486" alt="Schermafbeelding 2022-04-10 om 14 32 09" src="https://user-images.githubusercontent.com/33354141/162618293-68f7a4b9-2df9-46a3-9e3b-e2efd5a457e6.png">

which makes this functionality drift even more away from a Streamlined experience.

**Comment by ristomatti** [comment]
Please prioritize or at least confirm this will be fixed at some point. I was very excited about the entity id autocomplete in UI editors but this seems oddly to contradicting to what it offers. For me this is a blocker to upgrade. Especially the entity search, it's got to be a bug?

**Comment by emufan** [comment]
As the

- ids are now more or less hidden/unusable,
- the 2nd level config pages are a lot of clicks further away,
- the state set window in developer tools is hidden,
- the developer tools itself will be moved/hidden in whatever deep menu inside config menu in the next config menu release,
- the new cards (e.g. energy ones) don't have any customizing anymore
- etc. etc.
 
I unfortunately see that all this belongs together. In which direction, ever. Either the ones, who are deciding and implementing this, are not using it on their own, so they perhaps absolutely don't care about the problems and needed effort or perhaps don't want that other are using this anymore in the future at all. It is really a pity, as HA is of course great.

**Comment by KennethLavrsen** [comment]
I agree here. Hiding entity_id results in trouble for me also as I have duplicate names in many cases - at least until I rename the devices or entities.  We need to be able to see and look for entitiy IDs for many of the more advanced things and for finding why things do not work

**Comment by ristomatti** [comment]
> I unfortunately see that all this belongs together.

I can definitely imagine the long time goal would be to simplfy things for wider adoption. However I think it would be very premature to start hiding the technical details at this point. Attracted by the GUI configurability, I shifted from a purely Node-RED based setup ~6 months ago but I'm already neck deep in yaml. Without it, you'll only get so far.

I believe this change was just an unfortunate side-effect of trying to improve the UX for beginners. The team seems to be pushing new features at such a crazy pace, there's no way to avoid creating a ton of regression bugs given the massiven scale and complexity of the project.

But this discussion is not really contributing to solving the issue. Let's just hope it gets fixed sooner than later! :slightly_smiling_face:

**Comment by Mariusthvdb** [comment]
here's what Petro suggest to do about that: https://community.home-assistant.io/t/2022-4-groups-groups-groups/408965/579

we ned to take this from 'Issue' to 'Discussion'. 
suggestion to open:

> My recommendation is to do the following for a discussion:
> 
> Make all searches behave the same way. I.e. filter based on name or entity_id.
> When the result is selected from the searched list. It will show both the name and entity_id. Just like it does in the dropdown list when selecting.
> The visible name will continue to be the searchbox, where the entity_id will be static text that is not alterable. However you can highlight and select it.
> 

I believe that to be sane advise, and will try to do that lateron. If anyone else gets there before me, please dont hold back ;-) just reference this issue, to keep things connected

**Comment by ildar170975** [comment]
My suggestion is short - make it as it was before.
I do not appreciate a “make things simple and clear for dumb people” trend.

**Comment by zsarnett** [comment]
We will look at adding the entity ID back to the developer tools as we agree it should be there. But adding entity ID back to all entity dropdowns is not likely.

Filtering based on the ID could be possible for all dropdowns but will need to look at that implementation.

**Comment by Mariusthvdb** [comment]
openende discussion at https://github.com/home-assistant/frontend/discussions/12304

**Comment by bschatzow** [comment]
I agree with you @Mariusthvdb that all should be consistent. It looks according to @zsarnett  above that is not going to happen.

**Comment by Mariusthvdb** [comment]
hmm, nt directly related, but still: https://github.com/home-assistant/frontend/pull/12355
its going to be a whole other set of dev tools soon, where we get Tips too...

**Comment by emufan** [comment]
Where do you see "whole other set of dev tools soon"? I see only another (for me personally unwanted, because I know this for years) Tip appearing.

**Comment by Mariusthvdb** [comment]
see: https://github.com/home-assistant/frontend/discussions/12022#discussioncomment-2627542

**Comment by Mariusthvdb** [comment]
Keep

**Comment by Anders-McLean** [comment]
BUMP!! It has gone 7 months of pure pain and suffering. This bug is causing disasters every day of trying to develop for HA. PLEACE BRING BACK ENTITY IDS.  [Here](https://community.home-assistant.io/t/bring-back-entity-id-to-developer-tools/469727) is my longer rant about this.

**Comment by Mariusthvdb** [comment]
I've filed a WHT on this matter, because its also the other parts in the UI suffering this: https://community.home-assistant.io/t/wth-why-are-all-entities-named-identically-and-unrecognizable-in-the-ui-helper-options/467983

![8875fa2b07adf66ff9f881dccd0a704bd75ee8e6_2_419x500](https://user-images.githubusercontent.com/33354141/193996279-c7774d84-b290-4bbc-84f0-d286db2f9f9f.png)

**Comment by ildar170975** [comment]
Registered a similar issue for "Automations" list:
https://github.com/home-assistant/frontend/issues/14874

**Comment by ildar170975** [comment]
Finally solved in https://github.com/home-assistant/frontend/pull/16484

**Comment by Mariusthvdb** [comment]
yes, but only there...

not eg in Area overview:

<img width="361" alt="Scherm­afbeelding 2023-06-10 om 00 13 29" src="https://github.com/home-assistant/frontend/assets/33354141/e938bb7e-836c-4c99-8b53-43239271196a">

as mentioned [above](https://github.com/home-assistant/frontend/issues/11938#issuecomment-1060755498)

**Comment by ildar170975** [comment]
The original issue was only about Dev tools -> set state
Probably for other places separate issues required.

**Comment by kerryland** [comment]
> The original issue was only about Dev tools -> set state Probably for other places separate issues required.

It needs a holistic solution, not a piecemeal one.

**Comment by ildar170975** [comment]
up

**Comment by kerryland** [comment]
It's still really stupid

**Comment by ildar170975** [comment]
The initial issue is solved:
![image](https://github.com/home-assistant/frontend/assets/71872483/44b62690-a28e-4bb0-b056-85711557b54a)
As I already proposed before - for other places separated issues should be made.

**Comment by kerryland** [comment]
> As I already proposed before - for other places separated issues should be made.

And as I already suggested, this needs a holistic solution, not a piecemeal one. Let's not play whack-a-mole.

**Comment by kerryland** [comment]
i.e. if every page's behaviour is unique, I'd suggest they're doing it wrong.

**Comment by Mariusthvdb** [comment]
keep please.

**Comment by Mariusthvdb** [comment]
any point in keeping this open?

**Comment by ildar170975** [comment]
Do not think there is a reason of keeping THIS issue open.
For other places there are separate issues....

**Comment by ildar170975** [comment]
Hmm, this issue - https://github.com/home-assistant/frontend/issues/11870 - is about showing `friendly_name` in Automatuion UI editor.
But there are NO similar issues for:
1. UI editor (like for Entities card) - guess which entity is shown here (all are different ids):
![изображение](https://github.com/user-attachments/assets/a96de303-c65f-4b63-9410-49ef88637f07)

2. History page, Services (now Actions) - for "Target":
![изображение](https://github.com/user-attachments/assets/4b286b11-7d40-41da-8147-b689da4f71d3)
(there is only my ignored discussion - https://github.com/home-assistant/frontend/discussions/18640)

**Comment by kerryland** [comment]
Literally *everywhere* where "friendly name" is displayed it should be possible to easily view the entity_id, which is why I've been moaning about a "holistic solution". Custom code on every page is the wrong way to do it. 

But I *swear* I will shut up now and go away. No need to waste everyone's time listening to me!

**Comment by ildar170975** [comment]
> Literally everywhere where "friendly name" is displayed it should be possible to easily view the entity_id

Agreed

> which is why I've been moaning about a "holistic solution". Custom code on every page is the wrong way to do it.

Agreed again. But seems Devs do not care.

**Comment by Mariusthvdb** [comment]
keep, still very much desired

**Comment by Mariusthvdb** [comment]
....
so keep

**Comment by Mincka** [comment]
Maybe there's hope [in this recently merged PR](https://github.com/home-assistant/frontend/pull/24971).

![Image](https://github.com/user-attachments/assets/54036080-b729-4277-95f2-ae9cc336aae7)

![Image](https://github.com/user-attachments/assets/32a1768a-58ba-4789-8b90-c21e0afa267e)

We'll see next week in the beta.

**Comment by Mariusthvdb** [comment]
yes, if that would allow us to search by entity_id, that would be really nice, and very very long awaited ;-)

**Comment by ildar170975** [comment]
But a name of entity_id should be fully shown.

**Comment by silamon** [comment]
The pull request applied it to the entity picker, and the entity picker is also used in the dev tools states.
